internal/db/lens.go

```diff
@@ -13,22 +13,68 @@ package db
 import (
 	"context"
 
+	"github.com/ipfs/go-cid"
+
 	"github.com/sourcenetwork/corekv"
 	"github.com/sourcenetwork/immutable"
+	"github.com/sourcenetwork/immutable/enumerable"
+	"github.com/sourcenetwork/lens/host-go/config/model"
+	"github.com/sourcenetwork/lens/host-go/store"
 
 	"github.com/sourcenetwork/defradb/client"
 	"github.com/sourcenetwork/defradb/errors"
+	"github.com/sourcenetwork/defradb/internal/datastore"
 	"github.com/sourcenetwork/defradb/internal/db/description"
 )
 
-func (db *DB) setMigration(ctx context.Context, cfg client.LensConfig) error {
+// WARNING - This function does not create a txn ctx if one does not exist
+// and so should not be exposed to users.
+func (db *DB) AddLens(ctx context.Context, cfg model.Lens) (cid.Cid, error) {
```

Comment:
todo: This function is never used outside of this package. Instead of a warning not to expose it to users, you should simply make it private.